lib/models/message.ts

```diff
@@ -281,4 +293,52 @@ export class MessageModel extends DocumentDbModel<
       query: "SELECT * FROM messages m WHERE (m.fiscalCode = @fiscalCode)"
     });
   }
+
+  public async attachStoredContent(
+    blobService: BlobService,
+    messageId: string,
+    partitionKey: string,
+    message: IMessageWithContent
+  ): Promise<Either<Error, Option<DocumentDb.AttachmentMeta>>> {
+    // this is the attachment id __and__ the filename
+    const blobId = this.getMessageAttachmentName(messageId);
+
+    // store media (attachment) with message content in blob storage
+    const errorOrMessageContent = await upsertBlobFromText(
+      blobService,
+      MESSAGE_BLOB_STORAGE_CONTAINER_NAME,
+      blobId,
+      JSON.stringify({ id: messageId, ...message })
+    );
+
+    if (errorOrMessageContent.isLeft) {
+      return left(errorOrMessageContent.left);
+    }
+
+    // attach the created media to the message identified by messageId and partitionKey
+    const errorOrAttachmentMeta = await this.attach(messageId, partitionKey, {
+      contentType: "application/json",
+      id: blobId,
+      media: getBlobUrl(
+        blobService,
+        MESSAGE_BLOB_STORAGE_CONTAINER_NAME,
+        blobId
+      )
+    });
+
+    if (errorOrAttachmentMeta.isLeft) {
+      return left(
+        new Error(
+          `Error while attaching stored message: ${errorOrAttachmentMeta.left
+            .code} - ${errorOrAttachmentMeta.left.body}`
+        )
+      );
+    }
+
+    return right(errorOrAttachmentMeta.right);
+  }
+
+  private getMessageAttachmentName(id: string): string {
```

Comment:
this can be inlined since it's called only once, also it will make it more clear what is going on

Reply:
ok